addons/base/views.py

```diff
@@ -309,24 +309,33 @@ def get_auth(auth, **kwargs):
 ])
 
 
+def mark_file_version_as_seen(user, path):
+    file_to_update = OsfStorageFile.objects.get(_id=path)
+    # TODO - once the version is available in the waterbutler callback, update a specific version
```

Comment:
I think with [this commit](https://github.com/CenterForOpenScience/waterbutler/commit/81bec1ad3602177bca177a1ca0e8c93adc65b5e4) we can retrieve version info via query params used the request url. But It's a judgement call as to whether this is really in scope or not because a don't think anything about viewing intermediate versions is actually part of the ticket.

In either case, I have to grab the version this way as part of the upcoming analytics so one of is going to be piggybacking off the other.